src/main/java/edu/byu/ece/rapidSmith/design/subsite/CellDesign.java

```diff
@@ -188,49 +229,85 @@ private Cell registerCell(Cell cell) {
 
 		cell.setDesign(this);
 		cellMap.put(cell.getName(), cell);
+		
+		// add all internal nets when a macro is added to the design
+		if (cell.isMacro()) {
+			for (Cell internal : cell.getInternalCells()) {
+				internalCellMap.put(internal.getName(), internal);
+				internal.setDesign(this);
+			}
+			cell.getInternalNets().forEach(this::addNet);
+		}
+		
 		return cell;
 	}
 
 	/**
-	 * Disconnects and removes the specified cell from this design.
-	 *
-	 * The cell should not be a part of a molecule.
+	 * Disconnects and removes the specified cell from this design. For macro cells,
+	 * all internal nets and cells are also removed from the design. Internal cells
+	 * cannot be removed from the design, an exception will be thrown. Instead,
+	 * remove the parent macro cell. 
 	 *
 	 * @param cell the cell in this design to remove
 	 */
 	public void removeCell(Cell cell) {
 		Objects.requireNonNull(cell);
 		if (cell.getDesign() != this)
 			throw new Exceptions.DesignAssemblyException("Cannot remove cell not in the design.");
-
+		if (cell.isInternal()) 
+			throw new Exceptions.DesignAssemblyException("Cannot remove internal cell from the design. Remove the macro parent cell.");
+		
 		removeCell_impl(cell);
 	}
 
 	private void removeCell_impl(Cell cell) {
-		disconnectCell_impl(cell);
+		
 		cellMap.remove(cell.getName());
 		cell.clearDesign();
+		
+		// remove all of the internal cells and nets if a macro cell is removed
+		if (cell.isMacro()) {
+			for (Cell iCell: cell.getInternalCells()) {
+				iCell.clearDesign();
+				unplaceCell_impl(iCell);
```

Comment:
Not necessarily. It seems clunky as a user to have to manually unplace a cell and then remove it from the design in two separate steps. I would rather just call one method that unplaces the cell and removes it for me.